flow/visualize/i210_replay.py

```diff
@@ -226,17 +240,38 @@ def replay(args, flow_params, output_dir=None, transfer_test=None, rllib_config=
             for (key, lambda_func) in custom_callables.items():
                 custom_vals[key].append(lambda_func(env))
 
+            for past_veh_id in per_vehicle_energy_trace.keys():
+                if past_veh_id not in veh_ids:
+                    completed_vehicle_avg_energy[past_veh_id] = np.sum(per_vehicle_energy_trace[past_veh_id])
+                    completed_vehicle_travel_time[past_veh_id] = len(per_vehicle_energy_trace[past_veh_id])
+
+            for veh_id in veh_ids:
+                if veh_id not in initial_vehicles:
+                    if veh_id not in per_vehicle_energy_trace:
+                        # we have to skip the first step's energy calculation
+                        per_vehicle_energy_trace[veh_id].append(0)
+                    else:
+                        per_vehicle_energy_trace[veh_id].append(-1*vehicle_energy_consumption(env, veh_id))
+
             if type(done) is dict and done['__all__']:
                 break
             elif type(done) is not dict and done:
                 break
-
-        # Store the information from the run in info_dict.
-        outflow = env.k.vehicle.get_outflow_rate(int(500))
-        info_dict["velocities"].append(np.mean(vel))
-        info_dict["outflows"].append(outflow)
-        for key in custom_vals.keys():
-            info_dict[key].append(np.mean(custom_vals[key]))
+            elif max_completed_trips is not None and len(completed_vehicle_avg_energy) > max_completed_trips:
+                break
+        if env.crash:
+            print("Crash on iter", i)
```

Comment:
this is yet another hack that basically throws out a rollout if there was a crash. Crashes can wildly mess up the energy calculation

Reply:
@AboudyKreidieh this is where it's being used, essentially throws out an iteration if there was a crash, not sure if this is necessary, but it was needed to avoid getting weird energy calculations